Using default configuration in docker.service w/o MountFlags=slave flag, so you can see the same issue like bug 1347037.

Failing scenarios:
  features/rhelah/atomic/atomic_scanner.feature:45  8. Scanning rhel7 image
  features/rhelah/atomic/atomic_scanner.feature:51  9. Scanning all of images
  features/rhelah/atomic/atomic_scanner.feature:68  12. scan containers
  features/rhelah/atomic/atomic_scanner.feature:75  13. scan all of images and containers
  features/rhelah/atomic/atomic_scanner.feature:90  16. Remove all of containers
  features/rhelah/atomic/atomic_scanner.feature:94  17. Remove all of containers

0 features passed, 1 failed, 0 skipped
11 scenarios passed, 6 failed, 0 skipped
38 steps passed, 6 failed, 11 skipped, 0 undefined
Took 2m37.741s

NOTE: 16, 17 failed is caused by bug 1347037 ('xxx device or resource busy xxx'),  and there is no CVE bugs in the latest registry.access.redhat.com/rhel7 (sha256:5fbb7430), the test result as expected.

the details as follows.


$ behave features/rhelah/atomic/atomic_scanner.feature
Feature: Atomic scanner test # features/rhelah/atomic/atomic_scanner.feature:2
  Describes the basic 'atomic scan' command test
  Background: Atomic hosts are discovered  # features/rhelah/atomic/atomic_scanner.feature:5

  @list_default_scanners
  Scenario: 1. List default scanners             # features/rhelah/atomic/atomic_scanner.feature:9
    Given "all" hosts can be pinged              # steps/common.py:201 2.765s
    When List default scanners                   # steps/rhelah.py:714 1.148s
    Then find specified scanner "rhel7/openscap" # steps/rhelah.py:722 0.888s

  @pull_atomic_scan_openscap_image
  Scenario: 2. Pull latest openscap image from repository                             # features/rhelah/atomic/atomic_scanner.feature:14
    Given "all" hosts can be pinged                                                   # steps/common.py:201 0.208s
    When atomic update latest "docker.io/fedora/atomic_scan_openscap" from repository # steps/rhelah.py:534 5.995s
    Then Check whether "atomic_scan_openscap" is installed                            # steps/rhelah.py:543 0.872s

  @atomic_install_openscap_image
  Scenario: 3. Execute openscap image install method                    # features/rhelah/atomic/atomic_scanner.feature:19
    Given "all" hosts can be pinged                                     # steps/common.py:201 0.221s
    When Execute "docker.io/fedora/atomic_scan_openscap" install method # steps/rhelah.py:667 4.735s

  @list_current_scanners
  Scenario: 4. List available scanners                # features/rhelah/atomic/atomic_scanner.feature:23
    Given "all" hosts can be pinged                   # steps/common.py:201 0.223s
    When List available scanners                      # steps/rhelah.py:714 0.882s
    Then find specified scanner "rhel7/openscap"      # steps/rhelah.py:722 0.861s
    And find specified scanner "atomic_scan_openscap" # steps/rhelah.py:722 0.877s

  @pull_centos_image
  Scenario: 5. Pull latest centos image from repository  # features/rhelah/atomic/atomic_scanner.feature:29
    Given "all" hosts can be pinged                      # steps/common.py:201 0.215s
    When atomic update latest "centos" from repository   # steps/rhelah.py:534 11.291s
    Then Check whether "centos" is installed             # steps/rhelah.py:543 0.870s

  @scan_centos_image
  Scenario: 6. Scanning centos image                                     # features/rhelah/atomic/atomic_scanner.feature:34
    Given "all" hosts can be pinged                                      # steps/common.py:201 0.226s
    When Scan "centos" with "atomic_scan_openscap" and "cve" "--verbose" # steps/rhelah.py:729 9.007s
    Then Check "not supported" in scanner report                         # steps/rhelah.py:743 0.000s
    And check whether "/run/atomic" does not exist                       # steps/rhelah.py:527 0.221s

  @pull_rhel7_image
  Scenario: 7. Pull latest rhel7 image from repository                           # features/rhelah/atomic/atomic_scanner.feature:40
    Given "all" hosts can be pinged                                              # steps/common.py:201 0.202s
    When atomic update latest "registry.access.redhat.com/rhel7" from repository # steps/rhelah.py:534 62.118s
    Then Check whether "rhel7" is installed                                      # steps/rhelah.py:543 0.858s

  @scan_rhel7_image
  Scenario: 8. Scanning rhel7 image                                                                # features/rhelah/atomic/atomic_scanner.feature:45
    Given "all" hosts can be pinged                                                                # steps/common.py:201 0.212s
    When Scan "registry.access.redhat.com/rhel7" with "atomic_scan_openscap" and "cve" "--verbose" # steps/rhelah.py:729 8.709s
      Assertion Failed: {'192.168.122.134': {u'cmd': [u'atomic', u'scan', u'registry.access.redhat.com/rhel7', u'--scanner', u'atomic_scan_openscap', u'--scan_type', u'cve', u'--verbose'], u'end': u'2016-06-24 06:47:11.754195', u'stdout': u"INFO:OpenSCAP Daemon one-off evaluator 0.1.5
      WARNING:Can't import the 'docker' package. Container scanning functionality will be disabled.
      INFO:Creating tasks directory at '/var/lib/oscapd/tasks' because it didn't exist.
      INFO:Creating results directory at '/var/lib/oscapd/results' because it didn't exist.
      INFO:Creating results work in progress directory at '/var/lib/oscapd/work_in_progress' because it didn't exist.
      INFO:Evaluated EvaluationSpec, exit_code=0.
      INFO:Evaluated EvaluationSpec, exit_code=0.
      INFO:[100.00%] Scanned target 'chroot:///scanin/sha256:5fbb74308f2db1a6c2fa06e72ede9fdd4ecf34754e8e9280e382b77652b470a3'
      docker run -it --rm -v /etc/localtime:/etc/localtime -v /run/atomic/2016-06-24-06-47-03-889906:/scanin -v /var/lib/atomic/atomic_scan_openscap/2016-06-24-06-47-03-889906:/scanout:rw,Z -v /etc/oscapd:/etc/oscapd:ro docker.io/fedora/atomic_scan_openscap oscapd-evaluate scan --no-standard-compliance --targets chroots-in-dir:///scanin --output /scanout
      
      registry.access.redhat.com/rhel7 (sha256:5fbb7430)
      
      registry.access.redhat.com/rhel7 passed the scan
      
      Files associated with this scan are in /var/lib/atomic/atomic_scan_openscap/2016-06-24-06-47-03-889906.", u'changed': True, u'start': u'2016-06-24 06:47:03.269844', u'delta': u'0:00:08.484351', u'stderr': u'Error deleting container: Error response from daemon: Driver devicemapper failed to remove root filesystem 3f6c6863e52d9f343ee6222475ddfe39636f9b2fef531a7524f3b6356027d07c: remove /var/lib/docker/devicemapper/mnt/1fc9170efea2748229f0e661229133e240fda06b0ad193aeb0470581e663c89c: device or resource busy
      500 Server Error: Internal Server Error ("devmapper: Unknown device 62cd0ff580c042c450692a612d4ccd3d8003a789c31b41254428c398d876bfd5")', u'rc': 1, 'invocation': {'module_name': 'command', 'module_complex_args': {}, 'module_args': u'atomic scan registry.access.redhat.com/rhel7 --scanner atomic_scan_openscap --scan_type cve --verbose'}, u'warnings': []}}

Ansible Output: {'dark': {}, 'contacted': {'192.168.122.134': {u'cmd': [u'atomic', u'scan', u'registry.access.redhat.com/rhel7', u'--scanner', u'atomic_scan_openscap', u'--scan_type', u'cve', u'--verbose'], u'end': u'2016-06-24 06:47:11.754195', u'stdout': u"INFO:OpenSCAP Daemon one-off evaluator 0.1.5\r\nWARNING:Can't import the 'docker' package. Container scanning functionality will be disabled.\r\nINFO:Creating tasks directory at '/var/lib/oscapd/tasks' because it didn't exist.\r\nINFO:Creating results directory at '/var/lib/oscapd/results' because it didn't exist.\r\nINFO:Creating results work in progress directory at '/var/lib/oscapd/work_in_progress' because it didn't exist.\r\nINFO:Evaluated EvaluationSpec, exit_code=0.\r\nINFO:Evaluated EvaluationSpec, exit_code=0.\r\nINFO:[100.00%] Scanned target 'chroot:///scanin/sha256:5fbb74308f2db1a6c2fa06e72ede9fdd4ecf34754e8e9280e382b77652b470a3'\r\ndocker run -it --rm -v /etc/localtime:/etc/localtime -v /run/atomic/2016-06-24-06-47-03-889906:/scanin -v /var/lib/atomic/atomic_scan_openscap/2016-06-24-06-47-03-889906:/scanout:rw,Z -v /etc/oscapd:/etc/oscapd:ro docker.io/fedora/atomic_scan_openscap oscapd-evaluate scan --no-standard-compliance --targets chroots-in-dir:///scanin --output /scanout\n\nregistry.access.redhat.com/rhel7 (sha256:5fbb7430)\n\nregistry.access.redhat.com/rhel7 passed the scan\n\nFiles associated with this scan are in /var/lib/atomic/atomic_scan_openscap/2016-06-24-06-47-03-889906.", u'changed': True, u'start': u'2016-06-24 06:47:03.269844', u'delta': u'0:00:08.484351', u'stderr': u'Error deleting container: Error response from daemon: Driver devicemapper failed to remove root filesystem 3f6c6863e52d9f343ee6222475ddfe39636f9b2fef531a7524f3b6356027d07c: remove /var/lib/docker/devicemapper/mnt/1fc9170efea2748229f0e661229133e240fda06b0ad193aeb0470581e663c89c: device or resource busy\n500 Server Error: Internal Server Error ("devmapper: Unknown device 62cd0ff580c042c450692a612d4ccd3d8003a789c31b41254428c398d876bfd5")', u'rc': 1, 'invocation': {'module_name': 'command', 'module_complex_args': {}, 'module_args': u'atomic scan registry.access.redhat.com/rhel7 --scanner atomic_scan_openscap --scan_type cve --verbose'}, u'warnings': []}}}
    Then Check "issues were found" in scanner report                                               # None
    And check whether "/run/atomic" does not exist                                                 # None

  @scan_all_images
  Scenario: 9. Scanning all of images                                      # features/rhelah/atomic/atomic_scanner.feature:51
    Given "all" hosts can be pinged                                        # steps/common.py:201 0.215s
    When Scan "--images" with "atomic_scan_openscap" and "cve" "--verbose" # steps/rhelah.py:729 16.582s
      Assertion Failed: {'192.168.122.134': {u'cmd': [u'atomic', u'scan', u'--images', u'--scanner', u'atomic_scan_openscap', u'--scan_type', u'cve', u'--verbose'], u'end': u'2016-06-24 06:47:28.551411', u'stdout': u'INFO:OpenSCAP Daemon one-off evaluator 0.1.5
      WARNING:Can't import the 'docker' package. Container scanning functionality will be disabled.
      INFO:Creating tasks directory at '/var/lib/oscapd/tasks' because it didn't exist.
      INFO:Creating results directory at '/var/lib/oscapd/results' because it didn't exist.
      INFO:Creating results work in progress directory at '/var/lib/oscapd/work_in_progress' because it didn't exist.
      INFO:Evaluated EvaluationSpec, exit_code=0.
      ERROR:Failed to scan target 'chroot:///scanin/sha256:54552d1690f718e7acaf65c2eba88c3ded186c904ee6a230a8c7503b6bafe35e' for vulnerabilities.
      Traceback (most recent call last):
        File "/usr/bin/oscapd-evaluate", line 143, in scan_worker
          es.evaluate(config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 473, in evaluate
          wip_result = self.evaluate_into_dir(config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 470, in evaluate_into_dir
          return oscap_helpers.evaluate(self, config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 267, in evaluate
          args = get_evaluation_args(spec, config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 242, in get_evaluation_args
          ret.extend(spec.get_oscap_arguments(config))
        File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 444, in get_oscap_arguments
          ret.append(config.get_cve_feed(self.get_cpe_ids(config)))
        File "/usr/lib/python3.4/site-packages/openscap_daemon/config.py", line 385, in get_cve_feed
          return self.cve_feed_manager.get_cve_feed(cpe_ids)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/cve_feed_manager.py", line 225, in get_cve_feed
          "Can't find a supported CPE ID in %s" % (", ".join(cpe_ids))
      RuntimeError: Can't find a supported CPE ID in cpe:/o:centos:centos:7
      INFO:[ 25.00%] Scanned target 'chroot:///scanin/sha256:54552d1690f718e7acaf65c2eba88c3ded186c904ee6a230a8c7503b6bafe35e'
      INFO:Evaluated EvaluationSpec, exit_code=0.
      INFO:Evaluated EvaluationSpec, exit_code=0.
      ERROR:Failed to scan target 'chroot:///scanin/sha256:904d6c400333c03490f618654231344a9129cd4946049dc2878ce53558e7b816' for vulnerabilities.
      Traceback (most recent call last):
        File "/usr/bin/oscapd-evaluate", line 143, in scan_worker
          es.evaluate(config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 473, in evaluate
          wip_result = self.evaluate_into_dir(config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 470, in evaluate_into_dir
          return oscap_helpers.evaluate(self, config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 267, in evaluate
          args = get_evaluation_args(spec, config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 242, in get_evaluation_args
          ret.extend(spec.get_oscap_arguments(config))
        File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 444, in get_oscap_arguments
          ret.append(config.get_cve_feed(self.get_cpe_ids(config)))
        File "/usr/lib/python3.4/site-packages/openscap_daemon/config.py", line 385, in get_cve_feed
          return self.cve_feed_manager.get_cve_feed(cpe_ids)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/cve_feed_manager.py", line 225, in get_cve_feed
          "Can't find a supported CPE ID in %s" % (", ".join(cpe_ids))
      RuntimeError: Can't find a supported CPE ID in cpe:/o:centos:centos:7
      INFO:[ 50.00%] Scanned target 'chroot:///scanin/sha256:904d6c400333c03490f618654231344a9129cd4946049dc2878ce53558e7b816'
      INFO:Evaluated EvaluationSpec, exit_code=0.
      ERROR:Failed to scan target 'chroot:///scanin/sha256:a2b0a90bf7171b849a3226429f213ecf94fd638b1b3049fbe0c612ea91381093' for vulnerabilities.
      Traceback (most recent call last):
        File "/usr/bin/oscapd-evaluate", line 143, in scan_worker
          es.evaluate(config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 473, in evaluate
          wip_result = self.evaluate_into_dir(config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 470, in evaluate_into_dir
          return oscap_helpers.evaluate(self, config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 267, in evaluate
          args = get_evaluation_args(spec, config)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 242, in get_evaluation_args
          ret.extend(spec.get_oscap_arguments(config))
        File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 444, in get_oscap_arguments
          ret.append(config.get_cve_feed(self.get_cpe_ids(config)))
        File "/usr/lib/python3.4/site-packages/openscap_daemon/config.py", line 385, in get_cve_feed
          return self.cve_feed_manager.get_cve_feed(cpe_ids)
        File "/usr/lib/python3.4/site-packages/openscap_daemon/cve_feed_manager.py", line 225, in get_cve_feed
          "Can't find a supported CPE ID in %s" % (", ".join(cpe_ids))
      RuntimeError: Can't find a supported CPE ID in cpe:/o:fedoraproject:fedora:23
      INFO:[ 75.00%] Scanned target 'chroot:///scanin/sha256:a2b0a90bf7171b849a3226429f213ecf94fd638b1b3049fbe0c612ea91381093'
      INFO:Evaluated EvaluationSpec, exit_code=0.
      INFO:[100.00%] Scanned target 'chroot:///scanin/sha256:5fbb74308f2db1a6c2fa06e72ede9fdd4ecf34754e8e9280e382b77652b470a3'
      docker run -it --rm -v /etc/localtime:/etc/localtime -v /run/atomic/2016-06-24-06-47-12-817483:/scanin -v /var/lib/atomic/atomic_scan_openscap/2016-06-24-06-47-12-817483:/scanout:rw,Z -v /etc/oscapd:/etc/oscapd:ro docker.io/fedora/atomic_scan_openscap oscapd-evaluate scan --no-standard-compliance --targets chroots-in-dir:///scanin --output /scanout
      
      sha256:54552d1690f718e7acaf65c2eba88c3ded186c904ee6a230a8c7503b6bafe35e (sha256:54552d16)
      
           sha256:54552d1690f718e7acaf65c2eba88c3ded186c904ee6a230a8c7503b6bafe35e is not supported for this scan.
      
      sha256:904d6c400333c03490f618654231344a9129cd4946049dc2878ce53558e7b816 (docker.io/centos:latest)
      
           sha256:904d6c400333c03490f618654231344a9129cd4946049dc2878ce53558e7b816 is not supported for this scan.
      
      sha256:a2b0a90bf7171b849a3226429f213ecf94fd638b1b3049fbe0c612ea91381093 (docker.io/fedora/atomic_scan_openscap:latest)
      
           sha256:a2b0a90bf7171b849a3226429f213ecf94fd638b1b3049fbe0c612ea91381093 is not supported for this scan.
      
      sha256:5fbb74308f2db1a6c2fa06e72ede9fdd4ecf34754e8e9280e382b77652b470a3 (registry.access.redhat.com/rhel7:latest)
      
      sha256:5fbb74308f2db1a6c2fa06e72ede9fdd4ecf34754e8e9280e382b77652b470a3 passed the scan
      
      Files associated with this scan are in /var/lib/atomic/atomic_scan_openscap/2016-06-24-06-47-12-817483.', u'changed': True, u'start': u'2016-06-24 06:47:12.192328', u'delta': u'0:00:16.359083', u'stderr': u'500 Server Error: Internal Server Error ("devmapper: Unknown device 62cd0ff580c042c450692a612d4ccd3d8003a789c31b41254428c398d876bfd5")', u'rc': 1, 'invocation': {'module_name': 'command', 'module_complex_args': {}, 'module_args': u'atomic scan --images --scanner atomic_scan_openscap --scan_type cve --verbose'}, u'warnings': []}}

Ansible Output: {'dark': {}, 'contacted': {'192.168.122.134': {u'cmd': [u'atomic', u'scan', u'--images', u'--scanner', u'atomic_scan_openscap', u'--scan_type', u'cve', u'--verbose'], u'end': u'2016-06-24 06:47:28.551411', u'stdout': u'INFO:OpenSCAP Daemon one-off evaluator 0.1.5\r\nWARNING:Can\'t import the \'docker\' package. Container scanning functionality will be disabled.\r\nINFO:Creating tasks directory at \'/var/lib/oscapd/tasks\' because it didn\'t exist.\r\nINFO:Creating results directory at \'/var/lib/oscapd/results\' because it didn\'t exist.\r\nINFO:Creating results work in progress directory at \'/var/lib/oscapd/work_in_progress\' because it didn\'t exist.\r\nINFO:Evaluated EvaluationSpec, exit_code=0.\r\nERROR:Failed to scan target \'chroot:///scanin/sha256:54552d1690f718e7acaf65c2eba88c3ded186c904ee6a230a8c7503b6bafe35e\' for vulnerabilities.\r\nTraceback (most recent call last):\r\n  File "/usr/bin/oscapd-evaluate", line 143, in scan_worker\r\n    es.evaluate(config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 473, in evaluate\r\n    wip_result = self.evaluate_into_dir(config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 470, in evaluate_into_dir\r\n    return oscap_helpers.evaluate(self, config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 267, in evaluate\r\n    args = get_evaluation_args(spec, config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 242, in get_evaluation_args\r\n    ret.extend(spec.get_oscap_arguments(config))\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 444, in get_oscap_arguments\r\n    ret.append(config.get_cve_feed(self.get_cpe_ids(config)))\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/config.py", line 385, in get_cve_feed\r\n    return self.cve_feed_manager.get_cve_feed(cpe_ids)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/cve_feed_manager.py", line 225, in get_cve_feed\r\n    "Can\'t find a supported CPE ID in %s" % (", ".join(cpe_ids))\r\nRuntimeError: Can\'t find a supported CPE ID in cpe:/o:centos:centos:7\r\nINFO:[ 25.00%] Scanned target \'chroot:///scanin/sha256:54552d1690f718e7acaf65c2eba88c3ded186c904ee6a230a8c7503b6bafe35e\'\r\nINFO:Evaluated EvaluationSpec, exit_code=0.\r\nINFO:Evaluated EvaluationSpec, exit_code=0.\r\nERROR:Failed to scan target \'chroot:///scanin/sha256:904d6c400333c03490f618654231344a9129cd4946049dc2878ce53558e7b816\' for vulnerabilities.\r\nTraceback (most recent call last):\r\n  File "/usr/bin/oscapd-evaluate", line 143, in scan_worker\r\n    es.evaluate(config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 473, in evaluate\r\n    wip_result = self.evaluate_into_dir(config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 470, in evaluate_into_dir\r\n    return oscap_helpers.evaluate(self, config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 267, in evaluate\r\n    args = get_evaluation_args(spec, config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 242, in get_evaluation_args\r\n    ret.extend(spec.get_oscap_arguments(config))\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 444, in get_oscap_arguments\r\n    ret.append(config.get_cve_feed(self.get_cpe_ids(config)))\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/config.py", line 385, in get_cve_feed\r\n    return self.cve_feed_manager.get_cve_feed(cpe_ids)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/cve_feed_manager.py", line 225, in get_cve_feed\r\n    "Can\'t find a supported CPE ID in %s" % (", ".join(cpe_ids))\r\nRuntimeError: Can\'t find a supported CPE ID in cpe:/o:centos:centos:7\r\nINFO:[ 50.00%] Scanned target \'chroot:///scanin/sha256:904d6c400333c03490f618654231344a9129cd4946049dc2878ce53558e7b816\'\r\nINFO:Evaluated EvaluationSpec, exit_code=0.\r\nERROR:Failed to scan target \'chroot:///scanin/sha256:a2b0a90bf7171b849a3226429f213ecf94fd638b1b3049fbe0c612ea91381093\' for vulnerabilities.\r\nTraceback (most recent call last):\r\n  File "/usr/bin/oscapd-evaluate", line 143, in scan_worker\r\n    es.evaluate(config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 473, in evaluate\r\n    wip_result = self.evaluate_into_dir(config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 470, in evaluate_into_dir\r\n    return oscap_helpers.evaluate(self, config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 267, in evaluate\r\n    args = get_evaluation_args(spec, config)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/oscap_helpers.py", line 242, in get_evaluation_args\r\n    ret.extend(spec.get_oscap_arguments(config))\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/evaluation_spec.py", line 444, in get_oscap_arguments\r\n    ret.append(config.get_cve_feed(self.get_cpe_ids(config)))\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/config.py", line 385, in get_cve_feed\r\n    return self.cve_feed_manager.get_cve_feed(cpe_ids)\r\n  File "/usr/lib/python3.4/site-packages/openscap_daemon/cve_feed_manager.py", line 225, in get_cve_feed\r\n    "Can\'t find a supported CPE ID in %s" % (", ".join(cpe_ids))\r\nRuntimeError: Can\'t find a supported CPE ID in cpe:/o:fedoraproject:fedora:23\r\nINFO:[ 75.00%] Scanned target \'chroot:///scanin/sha256:a2b0a90bf7171b849a3226429f213ecf94fd638b1b3049fbe0c612ea91381093\'\r\nINFO:Evaluated EvaluationSpec, exit_code=0.\r\nINFO:[100.00%] Scanned target \'chroot:///scanin/sha256:5fbb74308f2db1a6c2fa06e72ede9fdd4ecf34754e8e9280e382b77652b470a3\'\r\ndocker run -it --rm -v /etc/localtime:/etc/localtime -v /run/atomic/2016-06-24-06-47-12-817483:/scanin -v /var/lib/atomic/atomic_scan_openscap/2016-06-24-06-47-12-817483:/scanout:rw,Z -v /etc/oscapd:/etc/oscapd:ro docker.io/fedora/atomic_scan_openscap oscapd-evaluate scan --no-standard-compliance --targets chroots-in-dir:///scanin --output /scanout\n\nsha256:54552d1690f718e7acaf65c2eba88c3ded186c904ee6a230a8c7503b6bafe35e (sha256:54552d16)\n\n     sha256:54552d1690f718e7acaf65c2eba88c3ded186c904ee6a230a8c7503b6bafe35e is not supported for this scan.\n\nsha256:904d6c400333c03490f618654231344a9129cd4946049dc2878ce53558e7b816 (docker.io/centos:latest)\n\n     sha256:904d6c400333c03490f618654231344a9129cd4946049dc2878ce53558e7b816 is not supported for this scan.\n\nsha256:a2b0a90bf7171b849a3226429f213ecf94fd638b1b3049fbe0c612ea91381093 (docker.io/fedora/atomic_scan_openscap:latest)\n\n     sha256:a2b0a90bf7171b849a3226429f213ecf94fd638b1b3049fbe0c612ea91381093 is not supported for this scan.\n\nsha256:5fbb74308f2db1a6c2fa06e72ede9fdd4ecf34754e8e9280e382b77652b470a3 (registry.access.redhat.com/rhel7:latest)\n\nsha256:5fbb74308f2db1a6c2fa06e72ede9fdd4ecf34754e8e9280e382b77652b470a3 passed the scan\n\nFiles associated with this scan are in /var/lib/atomic/atomic_scan_openscap/2016-06-24-06-47-12-817483.', u'changed': True, u'start': u'2016-06-24 06:47:12.192328', u'delta': u'0:00:16.359083', u'stderr': u'500 Server Error: Internal Server Error ("devmapper: Unknown device 62cd0ff580c042c450692a612d4ccd3d8003a789c31b41254428c398d876bfd5")', u'rc': 1, 'invocation': {'module_name': 'command', 'module_complex_args': {}, 'module_args': u'atomic scan --images --scanner atomic_scan_openscap --scan_type cve --verbose'}, u'warnings': []}}}
    Then Check "issues were found" in scanner report                       # None
    And Check "not supported" in scanner report                            # None
    And check whether "/run/atomic" does not exist                         # None

  @run_container1_in_bg
  Scenario: 10. docker run container1 with detach mode         # features/rhelah/atomic/atomic_scanner.feature:58
    Given "all" hosts can be pinged                            # steps/common.py:201 0.217s
    When docker run "centos" in detach mode with "C1" "top -b" # steps/docker.py:71 2.095s
    Then find latest created container by name "C1"            # steps/docker.py:84 0.275s

  @run_container2_in_bg
  Scenario: 11. docker run container2 with detach mode        # features/rhelah/atomic/atomic_scanner.feature:63
    Given "all" hosts can be pinged                           # steps/common.py:201 0.221s
    When docker run "rhel7" in detach mode with "C2" "top -b" # steps/docker.py:71 2.065s
    Then find latest created container by name "C2"           # steps/docker.py:84 0.285s

  @scan_containers
  Scenario: 12. scan containers                                                # features/rhelah/atomic/atomic_scanner.feature:68
    Given "all" hosts can be pinged                                            # steps/common.py:201 0.217s
    When Scan "--containers" with "atomic_scan_openscap" and "cve" "--verbose" # steps/rhelah.py:729 10.976s
      Assertion Failed: {'192.168.122.134': {u'cmd': [u'atomic', u'scan', u'--containers', u'--scanner', u'atomic_scan_openscap', u'--scan_type', u'cve', u'--verbose'], u'end': u'2016-06-24 06:47:44.909299', u'stdout': u'', u'changed': True, u'start': u'2016-06-24 06:47:34.160596', u'delta': u'0:00:10.748703', u'stderr': u'404 Client Error: Not Found ("open /dev/mapper/docker-253:0-9197922-a52559884515800e91d082b683d65718c2966afc90064b401be209dbfddec1bc: no such file or directory")', u'rc': 1, 'invocation': {'module_name': 'command', 'module_complex_args': {}, 'module_args': u'atomic scan --containers --scanner atomic_scan_openscap --scan_type cve --verbose'}, u'warnings': []}}

Ansible Output: {'dark': {}, 'contacted': {'192.168.122.134': {u'cmd': [u'atomic', u'scan', u'--containers', u'--scanner', u'atomic_scan_openscap', u'--scan_type', u'cve', u'--verbose'], u'end': u'2016-06-24 06:47:44.909299', u'stdout': u'', u'changed': True, u'start': u'2016-06-24 06:47:34.160596', u'delta': u'0:00:10.748703', u'stderr': u'404 Client Error: Not Found ("open /dev/mapper/docker-253:0-9197922-a52559884515800e91d082b683d65718c2966afc90064b401be209dbfddec1bc: no such file or directory")', u'rc': 1, 'invocation': {'module_name': 'command', 'module_complex_args': {}, 'module_args': u'atomic scan --containers --scanner atomic_scan_openscap --scan_type cve --verbose'}, u'warnings': []}}}
    Then Check "issues were found" in scanner report                           # None
    And Check "not supported" in scanner report                                # None
    And check whether "/run/atomic" does not exist                             # None

  @scan_images_and_containers
  Scenario: 13. scan all of images and containers                       # features/rhelah/atomic/atomic_scanner.feature:75
    Given "all" hosts can be pinged                                     # steps/common.py:201 0.220s
    When Scan "--all" with "atomic_scan_openscap" and "cve" "--verbose" # steps/rhelah.py:729 1.347s
      Assertion Failed: {'192.168.122.134': {u'cmd': [u'atomic', u'scan', u'--all', u'--scanner', u'atomic_scan_openscap', u'--scan_type', u'cve', u'--verbose'], u'end': u'2016-06-24 06:47:46.478174', u'stdout': u'', u'changed': True, u'start': u'2016-06-24 06:47:45.355254', u'delta': u'0:00:01.122920', u'stderr': u'500 Server Error: Internal Server Error ("devmapper: Error mounting '/dev/mapper/docker-253:0-9197922-991b0ccb471d9cedcd4b1d66f2f09cf1742bf2b4d73911ca82f7679ecfd1d1a0' on '/var/lib/docker/devicemapper/mnt/991b0ccb471d9cedcd4b1d66f2f09cf1742bf2b4d73911ca82f7679ecfd1d1a0': device or resource busy")', u'rc': 1, 'invocation': {'module_name': 'command', 'module_complex_args': {}, 'module_args': u'atomic scan --all --scanner atomic_scan_openscap --scan_type cve --verbose'}, u'warnings': []}}

Ansible Output: {'dark': {}, 'contacted': {'192.168.122.134': {u'cmd': [u'atomic', u'scan', u'--all', u'--scanner', u'atomic_scan_openscap', u'--scan_type', u'cve', u'--verbose'], u'end': u'2016-06-24 06:47:46.478174', u'stdout': u'', u'changed': True, u'start': u'2016-06-24 06:47:45.355254', u'delta': u'0:00:01.122920', u'stderr': u'500 Server Error: Internal Server Error ("devmapper: Error mounting \'/dev/mapper/docker-253:0-9197922-991b0ccb471d9cedcd4b1d66f2f09cf1742bf2b4d73911ca82f7679ecfd1d1a0\' on \'/var/lib/docker/devicemapper/mnt/991b0ccb471d9cedcd4b1d66f2f09cf1742bf2b4d73911ca82f7679ecfd1d1a0\': device or resource busy")', u'rc': 1, 'invocation': {'module_name': 'command', 'module_complex_args': {}, 'module_args': u'atomic scan --all --scanner atomic_scan_openscap --scan_type cve --verbose'}, u'warnings': []}}}
    Then Check "issues were found" in scanner report                    # None
    And Check "not supported" in scanner report                         # None
    And check whether "/run/atomic" does not exist                      # None

  @stop_container1
  Scenario: 14. atomic stop previous running container1  # features/rhelah/atomic/atomic_scanner.feature:82
    Given "all" hosts can be pinged                      # steps/common.py:201 0.224s
    When atomic stop container "C1"                      # steps/rhelah.py:472 2.591s

  @stop_container2
  Scenario: 15. atomic stop previous running container2  # features/rhelah/atomic/atomic_scanner.feature:86
    Given "all" hosts can be pinged                      # steps/common.py:201 0.240s
    When atomic stop container "C2"                      # steps/rhelah.py:472 3.585s

  @remove_all_containers
  Scenario: 16. Remove all of containers  # features/rhelah/atomic/atomic_scanner.feature:90
    Given "all" hosts can be pinged       # steps/common.py:201 0.216s
    When docker remove all of containers  # steps/docker.py:118 1.746s
      Traceback (most recent call last):
        File "/usr/lib/python2.7/site-packages/behave/model.py", line 1456, in run
          match.run(runner.context)
        File "/usr/lib/python2.7/site-packages/behave/model.py", line 1903, in run
          self.func(context, *args, **kwargs)
        File "steps/docker.py", line 122, in step_impl
          module_args='docker ps -a -q | xargs -r docker rm')
      AssertionError

Ansible Output: {'dark': {}, 'contacted': {'192.168.122.134': {u'cmd': u'docker ps -a -q | xargs -r docker rm', u'end': u'2016-06-24 06:47:55.084642', u'stdout': u'37f69453f3b0\nee3a1c815d6f\n74042c523b19\n45ed7bb6623e\n7dcf27b209ee\n4636408a6080\n4cb424a600ee\ncf237485f785', u'changed': True, u'start': u'2016-06-24 06:47:53.566394', u'delta': u'0:00:01.518248', u'stderr': u'Failed to remove container (46d737b16499): Error response from daemon: Driver devicemapper failed to remove root filesystem 46d737b16499724be0b32c00accb3d22ef6fbfc94c620d4dea72326005a38470: mount still active\nFailed to remove container (c616fffcdef6): Error response from daemon: Driver devicemapper failed to remove root filesystem c616fffcdef6bd7ca215a9f8e67d963ad3fe903639522e55b924c45dc476c926: mount still active\nFailed to remove container (3f6c6863e52d): Error response from daemon: Driver devicemapper failed to remove root filesystem 3f6c6863e52d9f343ee6222475ddfe39636f9b2fef531a7524f3b6356027d07c: remove /var/lib/docker/devicemapper/mnt/1fc9170efea2748229f0e661229133e240fda06b0ad193aeb0470581e663c89c: device or resource busy', u'rc': 123, 'invocation': {'module_name': 'shell', 'module_complex_args': {}, 'module_args': u'docker ps -a -q | xargs -r docker rm'}, u'warnings': []}}}

  @remove_all_images
  Scenario: 17. Remove all of containers  # features/rhelah/atomic/atomic_scanner.feature:94
    Given "all" hosts can be pinged       # steps/common.py:201 0.219s
    When docker remove all of images      # steps/docker.py:112 0.600s
      Traceback (most recent call last):
        File "/usr/lib/python2.7/site-packages/behave/model.py", line 1456, in run
          match.run(runner.context)
        File "/usr/lib/python2.7/site-packages/behave/model.py", line 1903, in run
          self.func(context, *args, **kwargs)
        File "steps/docker.py", line 116, in step_impl
          module_args='docker images -a -q | xargs -r docker rmi')
      AssertionError

Ansible Output: {'dark': {}, 'contacted': {'192.168.122.134': {u'cmd': u'docker images -a -q | xargs -r docker rmi', u'end': u'2016-06-24 06:47:55.903749', u'stdout': u'Deleted: sha256:515ccbec0e14094def1d6a4526290392186fd7be0d76808b297c9930815c281b\nDeleted: sha256:683cff610dc465a9df4ad45b639dc3079bb8cb4d7c86501bf2ae1abf0415a248\nDeleted: sha256:54552d1690f718e7acaf65c2eba88c3ded186c904ee6a230a8c7503b6bafe35e\nUntagged: registry.access.redhat.com/rhel7:latest\nDeleted: sha256:5fbb74308f2db1a6c2fa06e72ede9fdd4ecf34754e8e9280e382b77652b470a3\nDeleted: sha256:821a26b9542a1eb5c575a9f5095cb5c652a5f05b14b17d77e661ae1df7003a2b', u'changed': True, u'start': u'2016-06-24 06:47:55.525165', u'delta': u'0:00:00.378584', u'stderr': u'Failed to remove image (9fd969d366d3): Error response from daemon: conflict: unable to delete 9fd969d366d3 (must be forced) - image is being used by stopped container 46d737b16499\nFailed to remove image (a2b0a90bf717): Error response from daemon: conflict: unable to delete a2b0a90bf717 (must be forced) - image is being used by stopped container 3f6c6863e52d\nFailed to remove image (904d6c400333): Error response from daemon: conflict: unable to delete 904d6c400333 (cannot be forced) - image has dependent child images', u'rc': 123, 'invocation': {'module_name': 'shell', 'module_complex_args': {}, 'module_args': u'docker images -a -q | xargs -r docker rmi'}, u'warnings': []}}}


Failing scenarios:
  features/rhelah/atomic/atomic_scanner.feature:45  8. Scanning rhel7 image
  features/rhelah/atomic/atomic_scanner.feature:51  9. Scanning all of images
  features/rhelah/atomic/atomic_scanner.feature:68  12. scan containers
  features/rhelah/atomic/atomic_scanner.feature:75  13. scan all of images and containers
  features/rhelah/atomic/atomic_scanner.feature:90  16. Remove all of containers
  features/rhelah/atomic/atomic_scanner.feature:94  17. Remove all of containers

0 features passed, 1 failed, 0 skipped
11 scenarios passed, 6 failed, 0 skipped
38 steps passed, 6 failed, 11 skipped, 0 undefined
Took 2m37.741s

--- a/features/rhelah/atomic/atomic_scanner.feature
+++ b/features/rhelah/atomic/atomic_scanner.feature
@@ -1,0 +1,95 @@
+@atomic @host_subscribed @ah_upgrade
+Feature: Atomic scanner test
+    Describes the basic 'atomic scan' command test
+
+Background: Atomic hosts are discovered
+      Given "all" hosts can be pinged
+
+  @list_default_scanners
+  Scenario: 1. List default scanners
+       When List default scanners
+       Then find specified scanner "rhel7/openscap"
+
+  @pull_atomic_scan_openscap_image
+  Scenario: 2. Pull latest openscap image from repository
+       When atomic update latest "docker.io/fedora/atomic_scan_openscap" from repository
+       Then Check whether "atomic_scan_openscap" is installed
+
+  @atomic_install_openscap_image
+  Scenario: 3. Execute openscap image install method
+       When Execute "docker.io/fedora/atomic_scan_openscap" install method
+
+  @list_current_scanners
+  Scenario: 4. List available scanners
+       When List available scanners
+       Then find specified scanner "rhel7/openscap"
+        and find specified scanner "atomic_scan_openscap"
+
+  @pull_centos_image
+  Scenario: 5. Pull latest centos image from repository
+       When atomic update latest "centos" from repository
+       Then Check whether "centos" is installed
+
+  @scan_centos_image
+  Scenario: 6. Scanning centos image
+       When Scan "centos" with "atomic_scan_openscap" and "cve" "--verbose"
+       Then Check "not supported" in scanner report
+        and check whether "/run/atomic" does not exist
+
+  @pull_rhel7_image
+  Scenario: 7. Pull latest rhel7 image from repository
+       When atomic update latest "registry.access.redhat.com/rhel7" from repository
+       Then Check whether "rhel7" is installed
+
+  @scan_rhel7_image
+  Scenario: 8. Scanning rhel7 image
+       When Scan "registry.access.redhat.com/rhel7" with "atomic_scan_openscap" and "cve" "--verbose"
+       Then Check "issues were found" in scanner report
+        and check whether "/run/atomic" does not exist
+
+  @scan_all_images
+  Scenario: 9. Scanning all of images
+       When Scan "--images" with "atomic_scan_openscap" and "cve" "--verbose"
+       Then Check "issues were found" in scanner report
+        and Check "not supported" in scanner report
+        and check whether "/run/atomic" does not exist
+
+  @run_container1_in_bg
+  Scenario: 10. docker run container1 with detach mode
+       When docker run "centos" in detach mode with "C1" "top -b"
+       Then find latest created container by name "C1"
+
+  @run_container2_in_bg
+  Scenario: 11. docker run container2 with detach mode
+       When docker run "rhel7" in detach mode with "C2" "top -b"
+       Then find latest created container by name "C2"
+
+  @scan_containers
+  Scenario: 12. scan containers
+       When Scan "--containers" with "atomic_scan_openscap" and "cve" "--verbose"
+       Then Check "issues were found" in scanner report
+        and Check "not supported" in scanner report
+        and check whether "/run/atomic" does not exist
+
+  @scan_images_and_containers
+  Scenario: 13. scan all of images and containers
+       When Scan "--all" with "atomic_scan_openscap" and "cve" "--verbose"
+       Then Check "issues were found" in scanner report
+        and Check "not supported" in scanner report
+        and check whether "/run/atomic" does not exist
+
+  @stop_container1
+  Scenario: 14. atomic stop previous running container1
+       When atomic stop container "C1"
+
+  @stop_container2
+  Scenario: 15. atomic stop previous running container2
+       When atomic stop container "C2"
+
+  @remove_all_containers
+  Scenario: 16. Remove all of containers
+       When docker remove all of containers
+
+  @remove_all_images
+  Scenario: 17. Remove all of containers
+       When docker remove all of images

--- a/steps/rhelah.py
+++ b/steps/rhelah.py
@@ -128,6 +128,23 @@ def get_image_label(context, image, target='local'):
     return label_result
 
 
+def get_scanners(context):
+    '''Get available scanners'''
+    scanners = context.remote_cmd(cmd='command',
+                                  module_args='atomic scan --list')
+    assert not scanners is False, context.result['contacted']
+    return scanners[0]['stdout']
+
+
+def atomic_scanner(context, target, scanner, scan_type, options):
+    '''Scan images or containers or all'''
+    options = '--scanner %s --scan_type %s %s' % (scanner, scan_type, options)
+    scan_result = context.remote_cmd(cmd='command',
+                                     module_args='atomic scan %s %s' % (target, options))
+    assert not scan_result is False, context.result['contacted']
+    return scan_result[0]['stdout']
+
+
 @given(u'active tree version is at "{version}" on "{host}"')
 @then(u'active tree version is at "{version}" on "{host}"')
 def step_impl(context, version, host):
@@ -692,3 +709,48 @@ def step_impl(context, imageA, imageB, rpms='yes', arguments=''):
            assert "no file differences" not in result, same_image_msg
        else:
            assert result, diff_image_msg
+
+
+@when(u'List default scanners')
+@when(u'List available scanners')
+def step_impl(context):
+    '''List available scanners'''
+    scanners = get_scanners(context)
+    assert scanners, "Error can not list available scanners"
+
+
+@then(u'Find specified scanner "{scanner}"')
+def step_impl(context, scanner):
+    '''Find specified scanner'''
+    scanners = get_scanners(context)
+    assert scanner in scanners, "Can't find scanner '%s'" % scanner
+
+
+@when(u'Scan "{target}"')
+@when(u'Scan "{target}" with "{scanner}"')
+@when(u'Scan "{target}" with "{scanner}" and "{scan_type}"')
+@when(u'Scan "{target}" with "{scanner}" and "{scan_type}" "{options}"')
+def step_impl(context, target, scanner='openscap', scan_type='cve',
+    options='--verbose'):
+    '''Scan images or/and containers with scanner and scan type'''
+    scan_result = atomic_scanner(context, target, scanner, scan_type, options)
+    context.scan_result = scan_result
+    context.scan_type = scan_type
+
+    assert context.scan_result, "Error can't run atomic scan"
+
+
+@then(u'Check "{matches}" in scanner report')
+def step_impl(context, matches):
+    '''Check scanner result'''
+    result = context.scan_result
+    scan_type = context.scan_type
+    output = "Find '%s' in scanner report" % matches
+    if 'issues were found' in matches:
+        assert matches not in result, "Please file a '%s' type bug" % scan_type
+
+    elif 'not supported' in matches:
+        assert matches in result, "the image is not supported for this scan"
+
+    else:
+        assert matches not in result, output

